docs/spec/openapi/apm-openapi.yaml

```diff
@@ -0,0 +1,162 @@
+      operationId: getServerHealth
+      tags:
+        - server info
+      responses:
+        '200':
+          $ref: '#/components/responses/200_server_info'
+    post:
+      summary: Get general server information
+      operationId: postServerHealth
+      tags:
+        - server info
+      responses:
+        '200':
+          $ref: '#/components/responses/200_server_info'
+  /config/v1/agents:
+    get:
+      summary: Get agent configuration changes
+      description: >
+        To configure authenticated access to the APM server, the instructions at [APM API key](https://www.elastic.co/guide/en/observability/current/api-key.html) or [APM Secret Token](https://www.elastic.co/guide/en/observability/current/secret-token.html) must be followed to configure the correct permissions for APM access.
+      operationId: getAgentConfig
+      tags:
+        - agent config
+      parameters:
+        - in: query
+          name: service.name
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A successful response. 
+          content:
+            application/json:
+              schema:
+                type: object
+    post:
+      summary: Get agent configuration changes
+      description: >
+        To configure authenticated access to the APM server, the instructions at [APM API key](https://www.elastic.co/guide/en/observability/current/api-key.html) or [APM Secret Token](https://www.elastic.co/guide/en/observability/current/secret-token.html) must be followed to configure the correct permissions for APM access.
+      operationId: postAgentConfig
+      tags:
+        - agent config
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - service
+              properties:
+                CAPTURE_BODY:
+                  type: string
+                  example: off
+                service:
+                  type: object
+                  required:
+                    - name
+                  properties:
+                    environment:
+                      type: string
+                      example: all
+                    name:
+                      type: string
+                      example: test-service
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  capture_body:
+                   type: string
+                   example: off
+                  transaction_max_spans:
+                    type: integer
+                    example: 500
+                  transaction_sample_rate:
+                    type: number
+                    example: 0.3
```

Comment:
Is this supposed to be an exhaustive list? There are a lot more config options in https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/common/agent_configuration/setting_definitions/general_settings.ts

Reply:
It's only parity with what appears in the current docs, so definitely not an exhaustive list

Reply:
I've added `additionalProperties: true` to indicate there are more options than this currently covers. I've also pulled some descriptions from that file you linked, thanks!